ldk/javascript/src/whisper/form.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Components, Form, Whisper, WhisperComponentType } from '.';
+
+export class LdkForm {
+    children: Array<Components>;
+
+    componentState: Map<string, any>;
+
+    constructor(children: Array<Components>) {
+        this.componentState = new Map<string, any>();
+        this.children = children;
+
+        this.setInitialComponentState(this.children);
+        this.registerListeners(this.children);
+    }
+
+    getComponentState(): Map<string, any> {
```

Comment:
I was testing this further, and I noticed that there is a race condition when retrieving the current component state upon form submission. We are not guaranteed to get `onChange()` called before we retrieve the current state for each component collection.